### PR TITLE
Make all methods of dfcc_utilst static

### PR DESCRIPTION
**Pull request by tautschnig**
This class has no state, there is no need to instantiate it.

Depends on #7680, among others.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by remi-delmas-3000** [comment]
This PR seems to contain changes outside of `contracs/dynamic-frames`

**Comment by remi-delmas-3000** [comment]
We're in for a lot of refactoring if we go that way: all `dfcc_*` classes are actually singletons that keep references to context arguments that are present for the whole model transformation and would otherwise have to be passed explicitly to each and every function/method call:  the goto model, the symbol table, the message handler, the language mode, other singletons that provide utility functions, etc. 

It is usually a pain to realize you need a logger or a irep for the language mode and to have to modify a whole trail of function signatures just to propagate a context argument. 

Is there a better solution ?

**Comment by tautschnig** [comment]
> This PR seems to contain changes outside of `contracs/dynamic-frames`

Yes, that's #7680, which moves what was introduced as a DFCC utility to wider scope (and use).

**Comment by tautschnig** [comment]
> We're in for a lot of refactoring if we go that way: all `dfcc_*` classes are actually singletons that keep references to context arguments that are present for the whole model transformation and would otherwise have to be passed explicitly to each and every function/method call: the goto model, the symbol table, the message handler, the language mode, other singletons that provide utility functions, etc.
> 

While it was a good way to get us started, I am not sure this is an appropriate application of the Singleton pattern. We end up with many methods that have access to objects that they don't need to be able to access. This makes it very hard to see what a method does/does not change. Also, it's often not clear in what state a method expects a particular object to be in.

> It is usually a pain to realize you need a logger or a irep for the language mode and to have to modify a whole trail of function signatures just to propagate a context argument.
> 
> Is there a better solution ?

Not that I know of. Take #6915 as an example, where I had to thread through a message handler. Wasn't fun, but ultimately the change wasn't all that big. At the moment, we have a lot of maybe-we-need-this-later code, and I'm afraid I haven't yet seen that this would result in small changes/pull requests when new features are added.

File: src/goto-instrument/contracts/dynamic-frames/dfcc_utils.cpp [hunk 1]
```diff
@@ -309,15 +236,84 @@ void dfcc_utilst::clone_parameters(
...
+/// the given `function_id` function and transforming the existing
+/// function's name, parameter names, return type and source location
+/// using the given `trans_fun`, `trans_param` and `trans_ret_type` and
+/// `trans_loc` functions.
+///
+/// Also creates a new `goto_function` under the transformed name in
+/// the `function_map` with new parameters and an empty body.
+/// Returns the new symbol.
+///
+/// \param goto_model: target goto_model holding the symbol table
```
Comment (feliperodri) [inline src/goto-instrument/contracts/dynamic-frames/dfcc_utils.cpp:257]:
```suggestion
/// \param goto_model target goto_model holding the symbol table
```

**Review by qinheping — APPROVED** [review]
LGTM

**Review by feliperodri — APPROVED** [review]

File: src/goto-instrument/contracts/dynamic-frames/dfcc_utils.cpp (hunk 1 above)
Comment (tautschnig) [reply src/goto-instrument/contracts/dynamic-frames/dfcc_utils.cpp:257]: `CODING_STANDARD.md` suggests to have the colon here, so I'll keep it as-is.

**Pull request merged**